### PR TITLE
[bbr] implement new Sequence Number increase rules

### PR DESCRIPTION
**Pull request by edmont**
"BBR Sequence Number Updating" section in Thread Specification
requires special wraps when increasing BBR Dataset Sequence Number.

### CODE REVIEW

File: src/core/backbone_router/bbr_local.cpp [hunk 1]
```diff
@@ -52,7 +52,7 @@ Local::Local(Instance &aInstance)
     , mState(OT_BACKBONE_ROUTER_STATE_DISABLED)
     , mMlrTimeout(Mle::kMlrTimeoutDefault)
     , mReregistrationDelay(Mle::kRegistrationDelayDefault)
-    , mSequenceNumber(Random::NonCrypto::GetUint8())
+    , mSequenceNumber(Random::NonCrypto::GetUint8() % 127)
```
Comment (simonlingoogle) [inline src/core/backbone_router/bbr_local.cpp:55]: should we avoid 126 ?

**Review by simonlingoogle — COMMENTED** [review]
I wonder if we need THCI change, for code like `self.bbrSeqNum = random.randint(0, 254)`. 

@edmont

File: src/core/backbone_router/bbr_local.cpp (hunk 1 above)
Comment (edmont) [reply src/core/backbone_router/bbr_local.cpp:55]: What I understand from the specification is that 126 is possible as a random value.

Comment (simonlingoogle) [reply src/core/backbone_router/bbr_local.cpp:55]: Agree

File: tools/harness-thci/OpenThread.py [hunk 2]
```diff
@@ -1607,7 +1607,7 @@ def setDefaultValues(self):
             self.stopListeningToAddrAll()
 
         # BBR dataset
-        self.bbrSeqNum = random.randint(0, 254)  # random seqnum except 255, so that BBR-TC-02 never need re-run
+        self.bbrSeqNum = random.randint(0, 253)  # 5.21.4.2
```
[the remark this answers is not in the document]
Comment (edmont) [reply tools/harness-thci/OpenThread.py]: Yes. Done.

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1: 

Thanks

**Review by jwhui — APPROVED** [review]

**Pull request merged**